### DISCUSSION TITLE
[Feature Request] Allow negative budgets

### DISCUSSION THREAD

**Discussion by PsychoGame** [firefly-iii/firefly-iii#7507] [General]
I've spoken about this earlier, and you told me that it was intentional not to be able to have negative budgets.
I really like this software and would like to continue to use it for my personal finances, but for this I really need to be able to use negative budgets as well.
Each month I carry over budgets left over from the previous, with the addition of the divided salary income of the month.
Only when the amount of budget left-over from last month is negative, and the income is not sufficient my budget stays negative for that month (which indicates that I should pay more attention to that spending category).
Is it possible to bring this feature back (It worked in pre 6 versions of FireFly-III)
To show my gratitude I'm willing to become a Patreon of this project as well.

**Comment by JC5** [comment]
This will not be happening. However, I'm working on a feature where auto budgets will be decreased if you overspent the previous month: https://fosstodon.org/@ff3/110370786294569300

They will never drop below zero though.

**Reply by Zyadon7** [reply]
Can there be an option to allow budgets to drop below zero? I miss this feature and relied on it for any budget that overspent. In the current version .20 a negative balance gets converted to 1 which isn't consistent for amounts owed.

**Reply by JC5** [reply]
No. In the current version a negative budget only gets set to 1 if you consistently overspent the budget. You can always correct the budget yourself. Don't use budgets to track amounts owed, use liabilities instead.

**Reply by phi-sigsegv** [reply]
Why is the usage of budgets so pointlessly guarded by strange rules like that.
What one would expect as a behvaior is exactly what OP described: The budget could become negative.
Also technically this would be more than easy isn't it? Such a budget is just tracked by a single number and in auto budgets that number is increased by the configured amount, and expenses decrement it. Nothing more. No strange cut-offs at zero or whatever things like that. This is a behavior that's actually useful.
The current behavior is not useful at all. With a budget one tracks an amount of money that can be spend for a particular purpose. And If i have overspend that, then an automatic mechanism which cuts off at zero fools mento think that i have more money available to spend.
Sure one could track this with a liability in addition to the Budget, but that just adds an additional layer of complexity for no real gain.

So please please. Do not make things more complex than they have to be. Thanks. 🙂
